ui/src/app/core/settings.service.ts

```diff
@@ -0,0 +1,108 @@
+import {Injectable} from '@angular/core';
+import {AuthService} from '../core/auth.service';
+import {CapabilitiesService} from "./capabilities.service";
+import {DisplayField} from "../shared/model/DisplayField";
+
+type Settings = {
+  v1: {
+    // set to '' for basic auth users
+    userId: string
+    // set to '' for non-project-based job manager
+    projects: ProjectSettings[]
+  }
+};
+
+type ProjectSettings = {
+  projectId: string,
+  displayColumns: DisplayField[]
+  // ... other browser-based settings to be saved go here
+}
+
+@Injectable()
+export class SettingsService {
+  currentSettings: Settings;
+
+  /** If there are browser settings for the current user (or for '' if none) get them;
+   * otherwise, empty out the browser settings and set up a clean scaffolding */
+  constructor(
+    private readonly authService: AuthService,
+    private readonly capabilitiesService: CapabilitiesService,
+    private readonly localStorage: Storage
+  ) {
+    const savedSettings = JSON.parse(this.localStorage.getItem('settings'));
+    if ((authService.userId && savedSettings && (savedSettings.v1.userId == this.authService.userId))
+      || !authService.userId && savedSettings && savedSettings.v1.userId === '') {
+      this.currentSettings = savedSettings;
+    } else {
+      this.localStorage.setItem('settings', JSON.stringify(''));
+
+      let currentUser = '';
+      if (authService.userId) {
+        currentUser = this.authService.userId;
+      }
+      const projectSettings: ProjectSettings[] = [];
+      this.currentSettings = {
+        'v1': {
+          'userId' : currentUser,
+          'projects': projectSettings} };
+      this.updateLocalStorage();
+    }
+  }
+
+  /** return the display columns settings for a project, or if they don't exist, set all
+   * possible columns from capabilities service to be displayed and return that */
+  public getDisplayColumns(projectId = ''): DisplayField[] {
+    if (this.getSettingsForProject(projectId) && this.getSettingsForProject(projectId).displayColumns) {
+      return this.getSettingsForProject(projectId).displayColumns;
+    }
+    const cleanDisplayColumns = this.getDefaultDisplayFieldSettings();
+    this.currentSettings.v1.projects =[{
+      'projectId': projectId,
+      'displayColumns': cleanDisplayColumns
+    } as ProjectSettings];
+    this.updateLocalStorage();
```

Comment:
The reason is that if there are no saved settings, the UI needs something to go by.  Rather than generate defaults each time (if the user makes no changes), they are stored for future use.

Reply:
I think it's desirable to generate defaults each time. We shouldn't pin the user to the default columns that happened to be configured when they first logged in (unless they explicitly save that configuration).